### PR TITLE
[PREVIEW] - ROC-4359 Encrypt cookies

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4359

### Change description
Refactor cookies to use encryption

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/eligibility/store.ts [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-const minuteInMilliseconds = 60 * 1000
-const cookieTimeToLiveInMinutes = config.get<number>('eligibility.cookie.timeToLiveInMinutes') * minuteInMilliseconds
-
-export const cookieName = 'eligibility-check'
+export const cookieName = 'eligibilityCheck'
 
 export class CookieEligibilityStore {
   read (req: express.Request, res: express.Response): Eligibility {
-    const cookie: string = new Cookies(req, res).get(cookieName)
+    const cookie: string = req.signedCookies.eligibilityCheck
```
Comment (timja) [inline src/main/features/eligibility/store.ts]: will this just auto encrypt cookies that aren't encrypted ? or just make those users restart, chances are it'll be really low number of people that could be affected?

Comment (robertparkinson) [reply src/main/features/eligibility/store.ts]: it didn't like the - in the cookie name so I had to rename it. Yeah it should hopefully only effect a very small number of users if any at all.

Comment (timja) [reply src/main/features/eligibility/store.ts]: cookies page probably needs an update if the cookies been renamed

Comment (robertparkinson) [reply src/main/features/eligibility/store.ts]: Good spot! done.

File: infrastructure/main.tf [hunk 2]
```diff
@@ -32,6 +32,11 @@ data "azurerm_key_vault" "cmc_key_vault" {
   resource_group_name = "${local.vaultName}"
 }
 
+data "azurerm_key_vault_secret" "cookie_encryption_key" {
+  name = "citizen-cookie-encryption-key"
```
Comment (geek-soft) [inline infrastructure/main.tf]: @robertparkinson, I added secrets for AAT and prod.

File: src/main/features/eligibility/store.ts [hunk 3]
```diff
@@ -25,11 +24,10 @@ export class CookieEligibilityStore {
...
       return undefined
     }
-
-    new Cookies(req, res).set(cookieName, JSON.stringify(_.cloneDeepWith(eligibility, excludeDisplayValue)), { sameSite: 'lax', maxAge: cookieTimeToLiveInMinutes })
+    res.cookie(cookieName, _.cloneDeepWith(eligibility, excludeDisplayValue), { httpOnly: true, secure: true, sameSite: 'lax', maxAge: cookieTimeToLiveInMinutes })
   }
 
   clear (req: express.Request, res: express.Response): void {
-    new Cookies(req, res).set(cookieName, '', { sameSite: 'lax' })
+    res.clearCookie(cookieName, { sameSite: 'lax' })
```
Comment (geek-soft) [inline src/main/features/eligibility/store.ts]: @robertparkinson, setting empty value via `res.cookie` function does not remove the cookie so I had to use dedicated method.

File: src/main/features/eligibility/store.ts [hunk 4]
```diff
@@ -12,8 +11,8 @@ export const cookieName = 'eligibility-check'
 
 export class CookieEligibilityStore {
   read (req: express.Request, res: express.Response): Eligibility {
-    const cookie: string = new Cookies(req, res).get(cookieName)
-    return new Eligibility().deserialize(cookie !== undefined ? JSON.parse(cookie) : undefined)
+    const cookie: string = req.cookies[cookieName]
```
Comment (geek-soft) [inline src/main/features/eligibility/store.ts]: @robertparkinson, I disabled cookie signing as that does not bring much value and it complicates route testing.

**Pull request merged**